zcash_client_sqlite/src/lib.rs

```diff
@@ -457,21 +537,68 @@ impl<'a, P: consensus::Parameters> WalletWrite for DataConnStmtCache<'a, P> {
         })
     }
 
-    fn store_received_tx(
+    fn store_decrypted_tx(
         &mut self,
-        received_tx: &ReceivedTransaction,
+        d_tx: &DecryptedTransaction,
+        nullifiers: &[(AccountId, Nullifier)],
     ) -> Result<Self::TxRef, Self::Error> {
         self.transactionally(|up| {
-            let tx_ref = wallet::put_tx_data(up, received_tx.tx, None)?;
+            let tx_ref = wallet::put_tx_data(up, d_tx.tx, None)?;
 
-            for output in received_tx.outputs {
+            let mut spending_account_id: Option<AccountId> = None;
+            for output in d_tx.sapling_outputs {
                 if output.outgoing {
-                    wallet::put_sent_note(up, output, tx_ref)?;
+                    wallet::put_sent_note(
+                        up,
+                        tx_ref,
+                        output.index,
+                        output.account,
+                        &output.to,
+                        Amount::from_u64(output.note.value)
+                            .map_err(|_| SqliteClientError::CorruptedData("Note value invalid.".to_string()))?,
+                        Some(&output.memo),
+                    )?;
                 } else {
+                    match spending_account_id {
+                        Some(id) =>
+                            if id != output.account {
+                                panic!("Unable to determine a unique account identifier for z->t spend.");
+                            }
+                        None => {
+                            spending_account_id = Some(output.account);
+                        }
+                    }
+
                     wallet::put_received_note(up, output, tx_ref)?;
                 }
             }
 
+            // if we have some transparent outputs yet no shielded outputs, then this is t2t and we
```

Comment:
Why does it follow that a tx with some transparent outputs and no shielded outputs (which, given that there must be at least one output, is equivalent to "all transparent outputs") implies t2t? That seems wrong, maybe there is a typo.

Reply:
Maybe it was meant to say "shielded spends" instead of "shielded outputs".

Reply:
Looking at the code, yes `shielded spends` is what this should say.